app/assets/javascripts/app/helpers/text_formatter.js

```diff
@@ -1,146 +1,76 @@
+
+      url = // rebuild the url
+        (!addr.scheme ? '' : addr.scheme +
+        ( (addr.scheme.toLowerCase()=="mailto") ? ':' : '://')) +
+        (!addr.user ? '' : addr.user +
+        (!addr.pass ? '' : ':'+addr.pass) + '@') +
+        punycode.toASCII(addr.host) +
+        (!addr.port ? '' : ':' + addr.port) +
+        (!addr.path ? '' : encodeURI(addr.path) ) +
+        (!addr.query ? '' : '?' + encodeURI(addr.query) ) +
+        (!addr.fragment ? '' : '#' + encodeURI(addr.fragment) );
+      return url;
+    };
+
+    var md = window.markdownit({
+      breaks:      true,
+      html:        false,
+      linkify:     true,
+      typographer: true
     });
+    var iterator = window.markdownitForInline;
 
-    // make nice little utf-8 symbols
-    converter.hooks.chain("preConversion", function(text) {
-      var input_strings = [
-        "<->", "->", "<-",
-        "(c)", "(r)", "(tm)",
-        "<3"
-      ];
-      var output_symbols = [
-        "↔", "→", "←",
-        "©", "®", "™",
-        "♥"
-      ];
-      // quote function from: http://stackoverflow.com/a/494122
-      var quote = function(str) {
-        return str.replace(/([.?*+^$[\]\\(){}|-])/g, "\\$1");
-      };
-
-      _.each(input_strings, function(str, idx) {
-        var r = new RegExp(quote(str), "gi");
-        text = text.replace(r, output_symbols[idx]);
-      });
-      return text;
+    md.use(iterator, 'utf8_symbols', 'text', function (tokens, idx) {
+      tokens[idx].content = tokens[idx].content.replace(/<->/g, "↔")
+                                               .replace(/<-/g,  "←")
+                                               .replace(/->/g,  "→")
+                                               .replace(/<3/g,  "♥");
     });
 
-    converter.hooks.chain("postConversion", function (text) {
-      return text.replace(/(\"(?:(?:http|https):\/\/)?[a-zA-Z0-9\-\.]+\.[a-zA-Z]{2,3}(?:\/\S*)?\")(\>)/g, '$1 target="_blank">')
+    md.use(iterator, 'link_new_window_and_punycode', 'link_open', function (tokens, idx) {
+      tokens[idx].href = punycodeURL(tokens[idx].href);
+      tokens[idx].target = "_blank";
     });
 
-    return converter.makeHtml(text)
-  };
-
-  textFormatter.hashtagify = function hashtagify(text){
-    var utf8WordCharcters =/(<a[^>]*>.*?<\/a>)|(\s|^|>)#([\u0080-\uFFFF|\w|-]+|&lt;3)/g;
+    md.use(iterator, 'image_punycode', 'image', function (tokens, idx) {
+      tokens[idx].src = punycodeURL(tokens[idx].src);
+    });
 
-    return text.replace(utf8WordCharcters, function(result, linkTag, preceeder, tagText) {
-      if(linkTag)
-        return linkTag;
-      else
-        return preceeder + "<a href='/tags/" + tagText.toLowerCase() +
-                           "' class='tag'>#" + tagText + "</a>";
+    var hashtagPlugin = window.markdownitHashtag;
+    md.use(hashtagPlugin, {
+      // TODO is this [:alnum:] in ruby?
+      hashtagRegExp: '[\\x0080-\\xFFFF\\w\\-]+|<3',
```

Comment:
Can someone help here? I think it is important that we apply the markup to the same hashtags that `ActsAsTaggableOn` accepts. We set those accepted chars to `[:alnum:]` but I wasn't able to find out what exactly the chars in `[:alnum:]` are so I used the same values we had before.